File: Data Structures & Algorithms/house-robber/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    rob(nums) {
        let n = nums.length;
        if(n===1) return nums[0];
        let maxArr = [];
        maxArr[0] = nums[0];
        maxArr[1] = Math.max(nums[0],nums[1]);
        for(let i = 2 ;i < nums.length ; i++){
            maxArr[i]=Math.max(nums[i]+maxArr[i-2],maxArr[i-1]);
        }
        return maxArr[nums.length-1];
    }
    
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [1] * len(nums)

        for i in range(n-1,-1,-1):
            for j in range(i,n):
                if j <= n-1 and nums[j] > nums[i]:
                    dp[i] = max(dp[i], 1+dp[j])
        
        maxi = 0
        for i in range(n):
            maxi = max(maxi,dp[i])
        
        return maxi
